src/api/spec/controllers/labels_controller_spec.rb

```diff
@@ -0,0 +1,90 @@
+RSpec.describe LabelsController do
+  let(:user) { create(:confirmed_user, :with_home) }
+  let(:admin_user) { create(:admin_user, :with_home) }
+  let(:label_template) { create(:label_template, project: admin_user.home_project) }
+  let(:package) { create(:package, project: admin_user.home_project) }
+  let!(:label_admin) { create(:label, label_template: label_template, labelable: package) }
+
+  before do
+    Flipper.enable(:labels, user)
+    Flipper.enable(:labels, admin_user)
+  end
+
+  describe 'GET #index' do
+    render_views
+
+    context 'when labels are present' do
+      before do
+        login user
+
+        get :index, params: { project: admin_user.home_project.name, package: package.name, format: 'xml' }
+      end
+
+      it 'returns a list of labels' do
+        expect(Xmlhash.parse(response.body)['count']).to eq('1')
+        expect(Xmlhash.parse(response.body)['label']['id'].to_i).to eq(label_admin.id)
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    let(:new_package) { create(:package, project: admin_user.home_project) }
+    let(:xml) do
+      <<~XML
+        <label labelable_id="#{new_package.id}" labelable_type="Package" label_template_id="#{label_template.id}" />
+      XML
+    end
+
+    context 'when the user has the necessary permissions' do
+      before do
+        login admin_user
+        post :create, body: xml, params: { format: 'xml' }
+      end
+
+      it 'successfully creates a new label' do
+        expect(response).to have_http_status(:ok)
+        expect(new_package.labels.count).to eq(1)
```

Comment:
for the count I would use a subject block with the POST and then `expecrt { subject }.to change(Label, :count)`

Reply:
Done. I've added subject helper to make it more readable and DRY